app/src/main/java/com/example/poste/activities/LoginActivity.java

```diff
@@ -1,99 +1,127 @@
-import com.example.poste.api.poste.exceptions.APIException;
-import com.example.poste.api.poste.models.User;
+import com.example.poste.http.LoginRequest;
+import com.example.poste.http.MyApiService;
+import com.example.poste.http.RetrofitClient;
+import com.example.poste.models.User;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
-/**
- * The LoginActivity class adds functionality to the activity_login.xml layout
- */
 public class LoginActivity extends AppCompatActivity {
+
     public Button buttonLoginSubmit;
-    private Switch rememberMeSwitch;
-    private EditText usernameField,passwordField;
-    private String email;
-    private String password;
-    private User loginUser;
+    private EditText usernameField, passwordField;
 
     /**
-     * Called when the activity is created
-     *
-     * @param savedInstanceState A bundle containing the saved instance state
+     * function that runs on creation of the activity
+     * @param savedInstanceState
      */
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
-        // Configure window settings for fullscreen mode
-        requestWindowFeature(Window.FEATURE_NO_TITLE);
-        this.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
-                WindowManager.LayoutParams.FLAG_FULLSCREEN);
-        getSupportActionBar().hide();
-
-        // Set the activity layout
         setContentView(R.layout.activity_login);
 
-        // Prep vars
-        usernameField = findViewById(R.id.LoginEmailEntertxt);
-        passwordField = findViewById(R.id.LoginPassword);
-        rememberMeSwitch = findViewById(R.id.login_remember_me_switch);
-
-        //Email = findViewById(R.poste_item_id.LoginEmailEntertxt);
+        TextView hyperlinkTextView = findViewById(R.id.hyperlinkTextViewToRegister);
+        usernameField = findViewById(R.id.editTextTextEmailAddress);
+        passwordField = findViewById(R.id.editTextTextPassword);
 
-        //Page Navigation starts
-        buttonLoginSubmit = findViewById(R.id.LoginLoginbtn);
+        buttonLoginSubmit = findViewById(R.id.loginLoginbtn);
 
-        //Creating the link to Dashboard page
-        buttonLoginSubmit.setOnClickListener(view -> checkLogin());
-        //Page Navigation ends
-    }
+        /**
+         * adds a hyperlink to redirect user to the login page
+         */
+        SpannableString spannable = new SpannableString(hyperlinkTextView.getText());
+        ClickableSpan clickableSpan = new ClickableSpan() {
+            @Override
+            public void onClick(View widget) {
+                // Start the SecondActivity when the link is clicked
+                Intent intent = new Intent(LoginActivity.this, RegisterActivity.class);
+                startActivity(intent);
```

Comment:
I'm not 100% whether finish() should be called after we start new activities; something to keep in mind for the future:

https://stackoverflow.com/questions/10847526/what-is-activity-finish-method-doing-exactly

Reply:
Added finish() wherever I'm starting a new intent.